gzip/gzip.go

```diff
@@ -0,0 +1,83 @@
+package gzip
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io/ioutil"
+
+	kafka "github.com/segmentio/kafka-go"
+)
+
+func init() {
+	kafka.RegisterCompressionCodec(func() kafka.CompressionCodec {
+		return NewCompressionCodec()
+	})
+}
+
+type CompressionCodec struct {
+	CompressionLevel int
+}
+
+func NewCompressionCodec() CompressionCodec {
+	return NewCompressionCodecWith(kafka.DefaultCompressionLevel)
+}
+
+func NewCompressionCodecWith(level int) CompressionCodec {
+	return CompressionCodec{
+		CompressionLevel: level,
+	}
+}
+
+// Code implements the kafka.CompressionCodec interface.
+func (c CompressionCodec) Code() int8 {
+	return 1
+}
+
+// Encode implements the kafka.CompressionCodec interface.
+func (c CompressionCodec) Encode(dst, src []byte) (int, error) {
+	var buf bytes.Buffer
```

Comment:
Why use an intermediary buffer instead of an instance of your `buffer` type here? It causes an unnecessary copy where you could instead have the gzip writer output directly to the destination byte slice.